src/reactive/easyrsa.py

```diff
@@ -109,23 +109,31 @@ def configure_easyrsa():
 
 
 def configure_copy_extensions():
-    """Update the EasyRSA configuration with the capacity to copy the exensions
+    """Update the EasyRSA configuration with the capacity to copy the extensions
     through to the resulting certificates."""
-    # Create an absolute path to the file which will not be impacted by cwd.
-    (openssl_file,) = glob.iglob(os.path.join(easyrsa_directory, "openssl-*.cnf"))
+
+    openssl_confs = Path(easyrsa_directory).glob("openssl-*.cnf")
     # Update EasyRSA configuration with the capacity to copy CSR Requested
     # Extensions through to the resulting certificate. This can be tricky,
     # and the implications are not fully clear on this.
-    with open(openssl_file, "r") as f:
-        conf = f.readlines()
-    # When the copy_extensions key is not in the configuration.
-    if "copy_extensions = copy\n" not in conf:
-        for idx, line in enumerate(conf):
-            if "[ CA_default ]" in line:
-                # Insert a new line with the copy_extensions key set to copy.
-                conf.insert(idx + 1, "copy_extensions = copy\n")
-        with open(openssl_file, "w+") as f:
-            f.writelines(conf)
+
+    for openssl_file in openssl_confs:
+        conf = openssl_file.read_text().splitlines(True)
+        # When the copy_extensions key is not in the configuration.
+        if "copy_extensions = copy\n" not in conf:
+            for idx, line in enumerate(conf):
+                if "[ CA_default ]" in line:
+                    # Insert a new line with the copy_extensions key set to copy.
+                    conf.insert(idx + 1, "copy_extensions = copy\n")
+            openssl_file.write_text("\n".join(conf))
+
+        # Ensure the openssl-*.cnf file is copied down to the pki directory
+        # Remove any existing openssl-*.cnf files in the pki directory.
+        pki_path = openssl_file.parent / "pki"
+        if pki_path.exists():
+            for cnf in pki_path.glob("openssl-*.cnf"):
+                cnf.unlink()
+            shutil.copyfile(openssl_file, pki_path / openssl_file.name)
```

Comment:
This is a bit confusing for me. This fragment deletes all `openssl-*.cnf` files in the `pki` dir, then it copies just the one in the current iteration to that folder. By the time the loop finishes, only the last file remains in the `pki` dir, right? Am I missing something here?

Reply:
Is this safe because the charm should only manage one and only one config?

Reply:
Right, this deletes all `base/pki_path/openssl-*.cnf` and then copies `base/openssl-*.cnf` to `base/pki/`